test/all/ComponentHarness.mocha.js

```diff
@@ -0,0 +1,51 @@
+var expect = require('chai').expect;
+var testUtils = require('../../test-utils');
+var ComponentHarness = testUtils.ComponentHarness;
+
```

Comment:
Curious what happens if you have nested components like

```
Box/index.html

<index:>
  <view is="innerBox" />
  <view is="innerBox" />
```

if you call `getHtml`, would you get a shallow rendering so you don't also return the html of innerBox + it's children? 

Also curious if we want to add a helper to "find" a component inside a component.

Reply:
This is using the normal rendering, so there isn't a concept of shallow rendering. If for some reason you wanted a simpler mock, you could override the template for the inner views with an empty template:
```
harness.app.views.register('inner-box', '');
```

Is there a use case you had in mind for shallow rendering?

We typically use `as=` in the view to get a reference the inner component. If you have multiple inner elements you can do `as-object="map, key"` or `as-array="list"` as well.

Reply:
I was thinking to my past experiences and I've liked to shallow render because it helps to test *just* the parent component, rather than testing the parent component + assert on behaviour of all of its children. I guess it's really because react has a bunch of lifecycle methods and hooks but as I'm writing out this comment, i remember that derby doesn't act like react.

But - i think that registering children views as empty template is fine (its only wordy but nbd). My only concern is to be able to easily test out the structure of the parent + children!

Reply:
Yeah, I think mocking the children in the unit test is prudent; that's something we should at least have an example pattern for if not make the default. Sometimes you do want to test the full combination because it is part of the behavior you are testing, but you're right that you don't need to do that for most unit tests.

I need to add inter-component dependencies to this PR, so I can create examples of both as a next step.